### PR TITLE
Nutanix - content corrections (beta)

### PR DESCRIPTION
**Pull request by mihaitusa**
Added several beta operations.

### CODE REVIEW

File: content/io/cloudslang/nutanix/beta_tasks_poll.sl [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    - proxy_username:
+        required: false
+    - proxy_password:
+        required: false
+    - taskId
+
+  workflow:
+    - delete_vm:
+        do:
+          rest.http_client_get:
```
Comment (mihaitusa) [inline content/io/cloudslang/nutanix/beta_tasks_poll.sl]: @mbettan  You sure about this (use of http_client_get for a delete step)?

File: content/io/cloudslang/nutanix/beta_vms_clone.sl [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        required: false
+    - proxy_port:
+        default: "'8080'"
+        required: false
+    - proxy_username:
+        required: false
+    - proxy_password:
+        required: false
+    - templateId
+    - body
```
Comment (mihaitusa) [inline content/io/cloudslang/nutanix/beta_vms_clone.sl]: @mbettan  I think the body of a Post API call should not be present as input but constructed inside the http_client_post step.

File: content/io/cloudslang/nutanix/beta_vms_create.sl [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        required: false
+    - proxy_port:
+        default: "'8080'"
+        required: false
+    - proxy_username:
+        required: false
+    - proxy_password:
+        required: false
+    - templateId
+    - body
```
Comment (mihaitusa) [inline content/io/cloudslang/nutanix/beta_vms_create.sl]: @mbettan  Same comment as above.

**Comment by mihaitusa** [comment]
@gituser173 please review!

File: test/io/cloudslang/nutanix/test_nutanix_complete.sl [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        publish:
+          - return_result
+          - response
+          - error_message
+          - response_body: return_result
+
+    - print_reateResourceVMClone:
+        do:
+          print.print_text:
+            - text: response
```
Comment (gituser173) [inline test/io/cloudslang/nutanix/test_nutanix_complete.sl]: assert here? and in the other tests.

File: test/io/cloudslang/nutanix/test_nutanix_complete.sl [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        publish:
+          - return_result
+          - response
+          - error_message
+          - response_body: return_result
+
+    - print_nutanixCreateVM:
+        do:
+          print.print_text:
+            - text: response
```
Comment (gituser173) [inline test/io/cloudslang/nutanix/test_nutanix_complete.sl]: same comment as before

File: content/io/cloudslang/nutanix/beta_create_resource_vmclonedto.sl [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+####################################################
+# Creates an create.dto.acropolis.VMCloneDTO object.
+# For example, it could be used for clone a Virtual Machine operation.
+#
+# Inputs:
+#   - name - name for the cloned Virtual Machine
+#   - numVcpus - overrides the number of vCPUs assigned to the clone
+#   - memoryMb - overrides the amount of RAM assigned to the clone
```
Comment (gituser173) [inline content/io/cloudslang/nutanix/beta_create_resource_vmclonedto.sl]: snake case maybe? WDYT?

**Comment by oritstone** [comment]
We need to find a way to be able to test this content
For now leaving it on hold

**Comment by mbettan** [comment]
It would hard as Nutanix is hyperconverged infrastructure. Already tried to find a simulated API. Only way would be to work with Nutanix or their partner to get physical hardware. I had tested those operations through HP OO calls in customer site. Don't have access anymore to validate again.

**Comment by oritstone** [comment]
It is problematic for us to not be able to test it, since, for example, we just made some changes to our language structure (https://github.com/CloudSlang/cloud-slang/pull/476), so we already cannot verify that the content still works after changes like this...
We will continue to try to find a way to test that content, so we can add it to our repository.
If you happen to have access to such environment, or have an idea how it can be done, we will appreciate it a lot if you can assist in that.

**Comment by mbettan** [comment]
I will contact my customer to check if we can test remotely with him, but it will a one shoot test, we won't get a permanent access. What do you think? Who could be available with me for this remote call?

By the way, are we using in background some Service Virtualization technology? If we can capture those API calls once and can reproduce offsite that would be very efficient to validate clouds lang content.

**Comment by mbettan** [comment]
I think we will get more and more content without endpoint access for testing due to hard requirements or paying access. Moreover, any new release can break it. We really need to find a way to capture and simulate it offsite.

**Comment by mbettan** [comment]
Nutanix Prism Community Edition seams available, otherwise we will have to test the content with a partner/customer.

**Comment by mbettan** [comment]
Can we publish them as beta version?

**Comment by tethryus** [comment]
Pull Request closed until a better implementation with Nutanix will be available.

**Pull request closed without merging**